### PR TITLE
add depth limit on joinedload recursion

### PR DESCRIPTION
**Pull request by Idzikowski-Casey**

### CODE REVIEW

File: backend/sparrow/interface/schema.py [hunk 1]
```diff
@@ -93,30 +93,35 @@ def _nested_relationships(self, *, nests=None):
             - hybrid [default]: loads all models, attempting to pull back only needed
                 attributes that define the relationship...
         """
+        depth = current_depth
+        if depth > 1:
```
Comment (davenquinn) [inline backend/sparrow/interface/schema.py:97]: Maybe pass in max_depth as well....

File: backend/sparrow_tests/test_project_import.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def test_joined_load(self, db, statements):
         assert len(res) > 0
         output = ss.dump(res)
         assert len(output) == len(res)
-        assert len(statements) == 1
+        #assert len(statements) == 1 # statements is 80
```
Comment (davenquinn) [inline backend/sparrow_tests/test_project_import.py:87]: So wait, did you add a failing test, or did you make the failing test pass? Seems like the latter.

It'd be nice if you made depth configurable. Then we could still have this test pass but also configure the depth to 1 or 2 in application code.

**Review by davenquinn — CHANGES_REQUESTED** [review]
Make depth configurable

File: backend/sparrow/interface/schema.py (hunk 1 above)
Comment (Idzikowski-Casey) [reply backend/sparrow/interface/schema.py:97]: Okay, I will do that. A depth > 1 creates the same backend exiting we are seeing.

File: backend/sparrow_tests/test_project_import.py (hunk 2 above)
Comment (Idzikowski-Casey) [reply backend/sparrow_tests/test_project_import.py:87]: Agreed, I think we can make depth configurable at schema initialization.

Comment (davenquinn) [reply backend/sparrow_tests/test_project_import.py:87]: or better yet just make it configurable in the calling code, e.g., `schema.query_options(join_depth=2)`.

Comment (Idzikowski-Casey) [reply backend/sparrow_tests/test_project_import.py]: Okay, so now this test is failing, and the `max_depth` is configurable on the `schema.nested_relationships()` called in this test. So far changing the depth to a high number hasn't changed `len(statements)`.

**Review by davenquinn — CHANGES_REQUESTED** [review]
Can you make the default `max_depth=None` which doesn't limit recursion by default, but limit recursion to `max_depth=1` in the API? Then the test doesn't have to set an arbitrary max depth...

**Comment by davenquinn** [comment]
Once that last change is made you can go ahead and merge, assuming tests pass..

File: backend/sparrow/api/endpoints/model.py [hunk 3]
```diff
@@ -224,7 +224,7 @@ async def get(self, request):
         for _filter in self._filters:
             q = _filter(q, args)
 
-        q = q.options(*list(schema.query_options()))
+        q = q.options(*list(schema.query_options(max_depth=1)))
```
Comment (Idzikowski-Casey) [inline backend/sparrow/api/endpoints/model.py:227]: Set API to have a max_dept of 1

File: backend/sparrow/interface/schema.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ def __init__(self, *args, **kwargs):
     def _table(self):
         return self.opts.model.__table__
 
-    def _nested_relationships(self, *, nests=None, current_depth = 0, max_depth = 1):
+    def _nested_relationships(self, *, nests=None, current_depth=0, max_depth=None):
```
Comment (Idzikowski-Casey) [inline backend/sparrow/interface/schema.py:85]: max_depth has a default value of None

File: backend/sparrow/interface/schema.py [hunk 5]
```diff
@@ -114,24 +114,26 @@ def _nested_relationships(self, *, nests=None, current_depth = 0, max_depth = 1)
...
             field.schema.allowed_nests = _nests
             # Get relationships from nested models
-            for rel, load_hint in field.schema._nested_relationships(nests=_nests, current_depth = depth+1, max_depth=max_depth):
+            for rel, load_hint in field.schema._nested_relationships(
+                nests=_nests, current_depth=depth + 1, max_depth=max_depth
+            ):
                 yield [this_relationship, *rel], load_hint
 
-    def query_options(self, max_depth = 1):
+    def query_options(self, max_depth=None):
```
Comment (Idzikowski-Casey) [inline backend/sparrow/interface/schema.py:129]: max_depth is configurable from a few methods

**Pull request merged**